timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/BatchPaxosLearnerResource.java

```diff
@@ -39,14 +40,17 @@ public void learn(SetMultimap<Client, PaxosValue> paxosValuesByClient) {
     }
 
     @Override
-    public SetMultimap<Client, PaxosValue> getLearnedValues(Set<WithSeq<Client>> clientAndSeqs) {
+    public SetMultimap<Client, WithSeq<Optional<PaxosValue>>> getLearnedValues(Set<WithSeq<Client>> clientAndSeqs) {
         return KeyedStream.of(clientAndSeqs)
                 .mapKeys(WithSeq::value)
-                .map(WithSeq::seq)
-                .map((client, seq) -> paxosComponents.learner(client).getLearnedValue(seq))
+                .map(clientAndSeq -> clientAndSeq.map(this::getLearnedValue))
                 .collectToSetMultimap();
```

Comment:
I think if you want the old behaviour at this point you can filter out things which are paired with an empty